plugin/src/main/groovy/grails/plugin/springsecurity/acl/SpringSecurityAclGrailsPlugin.groovy

```diff
@@ -212,8 +211,6 @@ class SpringSecurityAclGrailsPlugin extends Plugin {
 
 	private configureExpressionBeans = { conf ->
 
-		parameterNameDiscoverer(ProxyAwareParameterNameDiscoverer)
```

Comment:
@jamesfredley @jdaugherty @codeconsole This PR has been approved but before I merge it there's one thing that I want to ask about because I am not sure who this is going to affect.  Once I got the tests running, it was clear there was an issue with this custom `ProxyAwareParameterNameDiscoverer`.   It just plain wasn't working: parameters weren't being discovered.  I will tag you in another line comment where this is being tested in the tests, which will make this more clear with an actual example of where parameter name discovery is needed.  It has to do with `@PreAuthorize` and other similar annotations.  

I am not sure why this behavior has changed between Spring (or Java) versions, but I think it has to do with the javac `-parameters` option.  The Spring [StandardReflectionParameterNameDiscoverer](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/StandardReflectionParameterNameDiscoverer.html) documentation alludes to this compiler parameter being required.  I don't think this compiler parameter is on by default.  I am not sure if that has changed with recent versions of Java, or perhaps it's a change in Spring behavior.

I don't think we want people to have to use compiler flags to get the old custom `ProxyAwareParameterNameDiscoverer` to work.  The alternative is to use the Spring `@P` annotation with Spring Security's [DefaultSecurityParameterNameDiscoverer](https://docs.spring.io/spring-security/site/docs/current/api//org/springframework/security/core/parameters/DefaultSecurityParameterNameDiscoverer.html) .  [@P is documented here in the Spring Security docs](https://docs.spring.io/spring-security/reference/servlet/authorization/method-security.html#using_method_parameters).  This is the route I decided to take.  I removed use of `ProxyAwareParameterNameDiscoverer`, deleted the class file, and modified the tests to use the `@P` annotation.  (The default `DefaultSecurityParameterNameDiscoverer` is in effect instead.)

Example from the docs:
```
@PreAuthorize("hasPermission(#c, 'write')")
public void updateContact(@P("c") Contact contact);
```

Issue #46 has been opened to update the docs.

If we don't want to go down this route, then I guess we would have to update the docs to tell people they have to use the `-parameters` compiler option and I can restore `ProxyAwareParameterNameDiscoverer`.

Reply:
Pinging @matrei for his thoughts too.